drivers/video/gc2145.c

```diff
@@ -1205,6 +1403,8 @@ static const struct gc2145_config gc2145_cfg_0 = {
 #if DT_INST_NODE_HAS_PROP(0, reset_gpios)
 	.reset_gpio = GPIO_DT_SPEC_INST_GET(0, reset_gpios),
 #endif
+	.bus_type = DT_PROP_OR(DT_CHILD(DT_INST_CHILD(0, port), endpoint), bus_type,
+			       VIDEO_BUS_TYPE_PARALLEL),
```

Comment:
It might be possible to use the macro `DT_INST_ENDPOINT_BY_ID(inst, pid, eid) for walking down the port and endpoint`:

https://github.com/zephyrproject-rtos/zephyr/blob/c35bb0de8023750de5ba31dcfbd7b5277466147e/include/zephyr/devicetree/port-endpoint.h#L105-L163